tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -274,6 +275,89 @@ def test_log_artifact_azure_blob_client_sas_error(self, databricks_artifact_repo
+    ):
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._get_write_credential_infos"
+        ) as write_credential_infos_mock, mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._azure_adls_gen2_upload_file"
+        ) as azure_upload_mock:
+            mock_credential_info = ArtifactCredentialInfo(
+                signed_uri=MOCK_ADL_GEN2_SIGNED_URI,
+                type=ArtifactCredentialType.AZURE_ADLS_GEN2_SAS_URI,
+            )
+            write_credential_infos_mock.return_value = [mock_credential_info]
+            azure_upload_mock.return_value = None
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credential_infos_mock.assert_called_with(
+                run_id=MOCK_RUN_ID, paths=[expected_location]
+            )
+            azure_upload_mock.assert_called_with(
+                mock_credential_info, test_file.strpath, expected_location
+            )
+
+    @pytest.mark.parametrize(("artifact_path", "expected_location"), [(None, "test.txt")])
+    def test_log_artifact_adl_gen2_with_headers(
+        self, databricks_artifact_repo, test_file, artifact_path, expected_location
+    ):
+        mock_azure_headers = {
+            "x-ms-content-type": "test-type",
+            "x-ms-owner": "some-owner",
+            "x-ms-blob-type": "some-type",
+        }
+        filtered_azure_headers = {
+            "x-ms-content-type": "test-type",
+            "x-ms-owner": "some-owner",
+        }
+        mock_response = Response()
+        mock_response.status_code = 200
+        mock_response.close = lambda: None
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._get_write_credential_infos"
+        ) as write_credential_infos_mock, mock.patch(
+            "mlflow.utils.rest_utils.cloud_storage_http_request"
+        ) as request_mock:
+            mock_credential_info = ArtifactCredentialInfo(
+                signed_uri=MOCK_ADL_GEN2_SIGNED_URI,
+                type=ArtifactCredentialType.AZURE_ADLS_GEN2_SAS_URI,
+                headers=[
+                    ArtifactCredentialInfo.HttpHeader(name=header_name, value=header_value)
+                    for header_name, header_value in mock_azure_headers.items()
+                ],
+            )
+            write_credential_infos_mock.return_value = [mock_credential_info]
+            request_mock.return_value = mock_response
+            databricks_artifact_repo.log_artifact(test_file.strpath, artifact_path)
+            write_credential_infos_mock.assert_called_with(
+                run_id=MOCK_RUN_ID, paths=[expected_location]
+            )
+            request_mock.assert_called_with(
+                "patch",
+                MOCK_ADL_GEN2_SIGNED_URI + "?action=flush&position=14",
+                headers=filtered_azure_headers,
+            )
+
+    def test_log_artifact_adl_gen2_blob_client_sas_error(self, databricks_artifact_repo, test_file):
```

Comment:
It seems that we don't need this because Gen 1 uses blob endpoint and Gen 2 uses dfs endpoint. Do we need a counterpart `test_log_artifact_adl_gen2_dfs_client_sas_error`?

Reply:
Modified test to return success for put and append and return error for flush.